internal/controllers/utils/conditions.go

```diff
@@ -84,3 +85,46 @@ func SetStatusCondition(existingConditions *[]metav1.Condition, conditionType Co
 		},
 	)
 }
+
+// IsStatusConditionPresent checks if the cluster provision condition is present
+func IsClusterProvisionPresent(cr *oranv1alpha1.ClusterRequest) bool {
+	condition := meta.FindStatusCondition(cr.Status.Conditions, (string(CRconditionTypes.ClusterProvisioned)))
+	return condition != nil
+}
+
+// IsClusterProvisionCompleted checks if the cluster provision condition status is completed
+func IsClusterProvisionCompleted(cr *oranv1alpha1.ClusterRequest) bool {
+	condition := meta.FindStatusCondition(cr.Status.Conditions, (string(CRconditionTypes.ClusterProvisioned)))
+	if condition != nil {
+		if condition.Status == metav1.ConditionTrue && condition.Reason == string(CRconditionReasons.Completed) {
+			return true
+		}
+	}
+	return false
+}
+
+// IsClusterProvisionTimedOutOrFailed checks if the cluster provision condition status is timedout or failed
+func IsClusterProvisionTimedOutOrFailed(cr *oranv1alpha1.ClusterRequest) bool {
+	condition := meta.FindStatusCondition(cr.Status.Conditions, (string(CRconditionTypes.ClusterProvisioned)))
+	if condition != nil {
+		if condition.Status == metav1.ConditionFalse &&
```

Comment:
Here it should be enough to only check the reasons.

Reply:
Yeah, but I prefer to also add the status false check to ensure that's expected behaviour - timeout/failed is false